test/e2e/e2e_test.go

```diff
@@ -0,0 +1,120 @@
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(k8sClient.Delete(ctx, ns)).To(gomega.Succeed())
+	})
+	ginkgo.When("Creating a Job without a matching LocalQueue", func() {
+		ginkgo.It("Should stay in suspended", func() {
+			lookupKey := types.NamespacedName{Name: "test-job", Namespace: ns.Name}
+			createdJob := &batchv1.Job{}
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+					return false
+				}
+				return *createdJob.Spec.Suspend
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+			createdWorkload := &kueue.Workload{}
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdWorkload); err != nil {
+					return false
+				}
+				return workload.InCondition(createdWorkload, kueue.WorkloadAdmitted)
+
+			}, framework.Timeout, framework.Interval).Should(gomega.BeFalse())
+			gomega.Expect(k8sClient.Delete(ctx, sampleJob)).Should(gomega.Succeed())
+		})
+	})
+	ginkgo.When("Creating A Job With Queueing", func() {
+		var (
+			resourceKueue *kueue.ResourceFlavor
+			localQueue    *kueue.LocalQueue
+			clusterQueue  *kueue.ClusterQueue
+		)
+		ginkgo.BeforeEach(func() {
+			resourceKueue = testing.MakeResourceFlavor("default").Obj()
+			gomega.Expect(k8sClient.Create(ctx, resourceKueue)).Should(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("main", ns.Name).Obj()
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor("default", "1").Obj()).Obj()).
+				Resource(testing.MakeResource(corev1.ResourceMemory).
+					Flavor(testing.MakeFlavor("default", "36Gi").Obj()).Obj()).Obj()
+			localQueue.Spec.ClusterQueue = "cluster-queue"
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(k8sClient.Delete(ctx, localQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Delete(ctx, clusterQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Delete(ctx, resourceKueue)).Should(gomega.Succeed())
+		})
+		ginkgo.It("Should unsuspend a job", func() {
+			lookupKey := types.NamespacedName{Name: "test-job", Namespace: ns.Name}
+			createdJob := &batchv1.Job{}
+			createdWorkload := &kueue.Workload{}
+
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+					return false
+				}
+				return !*createdJob.Spec.Suspend && createdJob.Status.Succeeded > 0
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdWorkload); err != nil {
+					return false
+				}
+				return workload.InCondition(createdWorkload, kueue.WorkloadAdmitted)
+
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
```

Comment:
You forgot to check that the Workload is marked as finished :)

Reply:
Good call.